src/main/java/org/terasology/behaviors/system/FindNearbyPlayersSystem.java

```diff
@@ -98,10 +84,7 @@ public void update(float delta) {
     }
 
     private boolean isEqual(List<EntityRef> one, List<EntityRef> two) {
-        if ((one == null && two != null) || (one != null && two == null)) {
-            return false;
-        }
-        if (one.size() != two.size()) {
+        if (one == null || two == null || one.size() != two.size()) {
             return false;
         }
```

Comment:
what is the rules for this comparison for equality. If either one of them are null it will hit a null reference exception when it checks the size of the two lists?

Reply:
yeah, when I first saw it I also thought it was strange. basically it checks (1) if either is null, and then (2) if the list sizes are the same, but if both lists are null you'll get an exception in (2). this condition, however, will never be triggered because of line 89 (and since this method is only used in line 92, I never bothered changing it until I completed the major changes in the module.